module/scheduler_components/per_key_remaster_manager.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <unordered_map>
+
+#include "module/scheduler_components/remaster_manager.h"
+
+using std::unordered_map;
+using std::unordered_set;
+using std::pair;
+
+
+namespace slog {
+
+/**
+ * Basic, inefficient implimentation of remastering. Transactions are kept in the exact
+ * order as their local logs: if a transaction from region 1 is blocked in the queue, all
+ * following transactions from region 1 will be blocked behind it
+ */
+class PerKeyRemasterManager :
+    public RemasterManager {
+public:
+  PerKeyRemasterManager(
+    shared_ptr<const Storage<Key, Record>> storage);
+
+  virtual VerifyMasterResult VerifyMaster(const TransactionHolder* txn_holder);
+  virtual RemasterOccurredResult RemasterOccured(Key key, uint32_t remaster_counter);
```

Comment:
Keys are strings so they should be passed with `const Key&` to avoid copying. Same goes for some of other functions below.